cmds/fspinfo/main.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017-2018 the LinuxBoot Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// fspinfo prints FSP header information.
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+
+	"github.com/linuxboot/fiano/pkg/fsp"
+	"github.com/linuxboot/fiano/pkg/uefi"
+)
+
+var (
+	flagJSON = flag.Bool("j", false, "Output as JSON")
+)
+
+func main() {
+	flag.Parse()
+	if flag.Arg(0) == "" {
+		log.Fatal("Error: missing file name")
+	}
+	data, err := ioutil.ReadFile(flag.Arg(0))
+	if err != nil {
+		log.Fatalf("Error: cannot read input file: %v", err)
+	}
+
+	fv, err := uefi.NewFirmwareVolume(data, 0, false)
+	if err != nil {
+		log.Fatalf("Error: cannot parse Firmware Volume: %v", err)
+	}
+	if len(fv.Files) < 1 {
+		log.Fatal("Error: Firmware Volume has no files")
+	}
+	file := fv.Files[0]
+	// FIXME why does FSPH start at +4?
+	hdr, err := fsp.FromBytes(file.Buf()[file.DataOffset+4 : file.DataOffset+4+fsp.HeaderV3Length])
+	if err != nil {
+		log.Fatalf("Error: cannot parse FSP Info Header: %v", err)
+	}
+	j, err := json.MarshalIndent(hdr, "", "    ")
+	if err != nil {
+		log.Fatalf("Error: cannot marshal JSON: %v", err)
+	}
+	if *flagJSON {
+		fmt.Println(string(j))
+	} else {
+		fmt.Printf("Signature                   : %s\n", hdr.Signature)
```

Comment:
If you need custom prints, could you make a stringer for the fsp header? rather than to do it here.